### PR TITLE
isa-l: add package

### PR DESCRIPTION
**Pull request by teepean**
Tested with 64-bit platforms

### CODE REVIEW

File: mingw-w64-isa-l/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+pkgdesc="A collection of optimized low-level functions targeting storage applications"
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64')
+url="https://github.com/intel/$pkgname"
+license=('BSD')
+depends=("${MINGW_PACKAGE_PREFIX}-libdeflate")
+makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
+             "${MINGW_PACKAGE_PREFIX}-autotools"
+             "${MINGW_PACKAGE_PREFIX}-yasm")
+source=("$url/archive/v$pkgver.tar.gz")
```
Comment (Biswa96) [inline mingw-w64-isa-l/PKGBUILD:16]: Redundant source array.

File: mingw-w64-isa-l/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,52 @@
+# Maintainer: Roy Oursler <roy.j.oursler@intel.com>
+_realname=isa-l
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=2.30.0
+pkgrel=1
+pkgdesc="A collection of optimized low-level functions targeting storage applications"
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64')
+url="https://github.com/intel/$pkgname"
```
Comment (Biswa96) [inline mingw-w64-isa-l/PKGBUILD:10]: Direct link should be used instead of using variable. Here pkgname variable is not same with upstream link.

**Comment by MehdiChinoune** [comment]
So you just copied the file from arch and modified it.
Please fellow PKGBUILD-templates.

File: mingw-w64-isa-l/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Maintainer: Roy Oursler <roy.j.oursler@intel.com>
```
Comment (MehdiChinoune) [inline mingw-w64-isa-l/PKGBUILD:1]: Are you Roy Oursler?

File: mingw-w64-isa-l/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# Contributor: Teemu Nätkinniemi ("teepean") <tnatkinn@gmail.com>
+_realname=isa-l
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=2.30.0
+pkgrel=1
+pkgdesc="A collection of optimized low-level functions targeting storage applications"
```
Comment (MehdiChinoune) [inline mingw-w64-isa-l/PKGBUILD:7]:
```suggestion
pkgdesc="A collection of optimized low-level functions targeting storage applications (mingw-w64)"
```

File: mingw-w64-isa-l/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+arch=('any')
+mingw_arch=('mingw64' 'ucrt64' 'clang64')
+url="https://github.com/intel/isa-l"
+license=('BSD')
+depends=("${MINGW_PACKAGE_PREFIX}-libdeflate")
+makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
+             "${MINGW_PACKAGE_PREFIX}-autotools"
+             "${MINGW_PACKAGE_PREFIX}-yasm")
+source=("${_realname}-${pkgver}.tar.gz::https://github.com/intel/${_realname}/archive/refs/tags/v${pkgver}.tar.gz")
+sha1sums=('80f55dd80ab06d5db6eb2d9ff38004ab3872a0f0')
```
Comment (MehdiChinoune) [inline mingw-w64-isa-l/PKGBUILD:17]: Please use a sha256sums

File: mingw-w64-isa-l/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+prepare(){
+  tar -xzvf "${srcdir}/${_realname}-${pkgver}.tar.gz"
```
Comment (MehdiChinoune) [inline mingw-w64-isa-l/PKGBUILD:20]:
```suggestion
```
Archives get extracted automatically, there is no need to add this step

File: mingw-w64-isa-l/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  cd "${srcdir}/${_realname}-${pkgver}"
+  # dont allow undefined symbols to build the dll
+  sed -i "87ilibisal_la_LDFLAGS = -no-undefined" Makefile.am
+  ./autogen.sh
+}
+
+build() {
+  [[ -d "build-${MINGW_CHOST}" ]] && rm -rf "build-${MINGW_CHOST}"
+  mkdir -p "${srcdir}/build-${MINGW_CHOST}"
+  cd "${srcdir}/build-${MINGW_CHOST}"
```
Comment (MehdiChinoune) [inline mingw-w64-isa-l/PKGBUILD:30]:
```suggestion
  mkdir -p "${srcdir}/build-${MSYSTEM}" && cd "${srcdir}/build-${MSYSTEM}"
```

File: mingw-w64-isa-l/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    --host=${MINGW_CHOST} \
+    --target=${MINGW_CHOST} \
+    --enable-static \
+    --enable-shared
+
+  make
+}
+
+check() {
+    cd "${srcdir}/build-${MINGW_CHOST}"
```
Comment (MehdiChinoune) [inline mingw-w64-isa-l/PKGBUILD:44]:
```suggestion
    cd "${srcdir}/build-${MSYSTEM}"
```

File: mingw-w64-isa-l/PKGBUILD [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    make check
+}
+
+package() {
+  cd "${srcdir}/build-${MINGW_CHOST}"
```
Comment (MehdiChinoune) [inline mingw-w64-isa-l/PKGBUILD:49]:
```suggestion
  cd "${srcdir}/build-${MSYSTEM}"
```

**Review by MehdiChinoune — CHANGES_REQUESTED** [review]

File: mingw-w64-isa-l/PKGBUILD (hunk 5 above)
Comment (teepean) [reply mingw-w64-isa-l/PKGBUILD]: Committed.

File: mingw-w64-isa-l/PKGBUILD (hunk 8 above)
Comment (teepean) [reply mingw-w64-isa-l/PKGBUILD:43]: This change seems to break most of the builds so I have reverted it.

File: mingw-w64-isa-l/PKGBUILD (hunk 7 above)
Comment (teepean) [reply mingw-w64-isa-l/PKGBUILD]: This is now fixed.

File: mingw-w64-isa-l/PKGBUILD [hunk 10]
```diff
@@ -24,27 +24,24 @@ prepare(){
...
-  cd "${srcdir}/build-${MINGW_CHOST}"
+  [[ -d "build-${MSYSTEM}" ]] && rm -rf "build-${MSYSTEM}"
+  mkdir -p "${srcdir}/build-${MSYSTEM}"
+  cd "${srcdir}/build-${MSYSTEM}"
 
   ../${_realname}-${pkgver}/configure \
     --prefix=${MINGW_PREFIX} \
-    --build=${MINGW_CHOST} \
-    --host=${MINGW_CHOST} \
-    --target=${MINGW_CHOST} \
```
Comment (MehdiChinoune) [inline mingw-w64-isa-l/PKGBUILD]: I didn't ask to remove these!

Comment (teepean) [reply mingw-w64-isa-l/PKGBUILD]: The Glib example PKGBUILD doesn't have those so I removed them.

File: mingw-w64-isa-l/PKGBUILD (hunk 9 above)
Comment (teepean) [reply mingw-w64-isa-l/PKGBUILD]: Fixed.

**Comment by teepean** [comment]
I will start over.

**Comment by MehdiChinoune** [comment]
Please, squash all commits and force-push

**Pull request closed without merging**